scripts/testing/e2e-runner

```diff
@@ -0,0 +1,348 @@
+# RESULT_ROOT:   directory for publishing results
+# RESULT_NAME:   subdirectory for publishing results of this test run
+# SKIP_WORKTREE: skip preparing worktree, assume it is already done
+# E2E_TESTS:     e2e tests to run
+
+REMOTE_REPO="${REMOTE_REPO:-https://github.com/containers/nri-plugins}"
+CLONED_REPO="${CLONED_REPO:-/opt/e2e-test/nri-plugins/nri-plugins}"
+TEST_BRANCH="${TEST_BRANCH:-main}"
+RESULT_ROOT="${RESULT_ROOT:-/opt/e2e-test/nri-plugins/results}"
+RESULT_NAME="${RESULT_NAME:-$(date +%Y-%m-%d-%H-%M)}"
+PROXIES_SRC=""
+SKIP_WORKTREE="${SKIP_WORKTREE:-}"
+E2E_TESTS="${E2E_TESTS:-}"
+
+
+# This script.
+SELF="$0"
+TEST_SUITE_DIR=policies.test-suite
+
+#
+# Derived configuration, set up after command line parsing
+#
+# RESULT_DIR: directory for publishing results of this test run
+# WORKTREE:   git worktree for this test run
+# LOG:        log file for this test run
+
+RESULT_DIR=""
+LOG=""
+WORKTREE=""
+
+usage() {
+    local exit_code="${exit_code:-0}"
+
+    echo "usage: ${1:-$SELF} [options] [<e2e-tests>]"
+    echo
+    echo "The possible options are:"
+    echo "  --source-proxies <file>  source proxy environment variables from file"
+    echo "  --remote|r <repo>        remote repository to clone/fetch from"
+    echo "  --branch|-b <branch>     name of the remote branch to test"
+    echo "  --local|-l <dir>         local directory to clone remote respository into"
+    echo "  --results|--out <dir>    top-level directory for publishing results"
+    echo "  --name|-n) <name>        subdirectory for results of this run"
+    echo "  --skip-worktree)         skip worktree preparation, assume already done"
+    echo "  --help)                  show this help"
+
+    exit $exit_code
+}
+
+info() {
+    echo 1>&2 "$@"
+}
+
+error() {
+    echo 1>&2 "error: $@"
+}
+
+fatal() {
+    echo 1>&2 "fatal error: $@"
+    exit 1
+}
+
+source_proxies() {
+    local var
+
+    if [ -z "$PROXIES_SRC" ]; then
+        info "no proxies to set up..."
+        return 0
+    fi
+
+    info "setting up proxies..."
+    if ! source "$PROXIES_SRC"; then
+        fatal "failed to source proxies from $src"
+    fi
+
+    for var in ${!http*} ${!HTTP*} ${!no_*} ${!NO_*}; do
+        case $var in
+            *_proxy|*_PROXY)
+                info "exporting proxy envrionment variable $var (with value ${!var})..."
+                export $var
+                ;;
+        esac
+    done
+}
+
+prepare_worktree() {
+    local worktree_dir="$(dirname $CLONED_REPO)/worktree"
+
+    WORKTREE="$worktree_dir/$RESULT_NAME"
+    case "$SKIP_WORKTREE" in
+        1|true|yes|y)
+            WORKTREE="$(realpath $WORKTREE)"
+            if ! cd $WORKTREE; then
+                fatal "failed to cd into worktree $WORKTREE..."
+            fi
+            return 0
+            ;;
+    esac
+
+    if [ ! -e "$CLONED_REPO" ]; then
+        if ! mkdir -p $CLONED_REPO; then
+            fatal "failed to create directory $CLONED_REPO"
+        fi
+        cd $CLONED_REPO
```

Comment:
probably good to check exit status of `cd` just like mkdir and git clone, too, and bail out in case of error. if someone else has created /tmp/repo and this user has no right to change into it, the script will continue and clone the repo in some surprising place.

the same applies to other places, too. should we have `cd-or-die()` or `must-cd()` or something to keep this short?

possibly also good idea double-quote $CLONED_REPO and others in commands, as at least shellcheck would complain about them.

Reply:
Updated. Introduced/used `must-{cd,mkdir,mv}`, and quoted variable instances in most places.